### ISSUE TITLE
url not JSON serializable

### ISSUE THREAD

**Issue by gubuntu** [inasafe/inasafe#4201]
After running an anlysis for 16h it failed with this error!

```
2017-05-23T15:38:13	0	ANALYSIS : Summary calculation
2017-05-23T15:38:13	0	impact function: Aggregate the earthquake analysis summary
2017-05-23T15:38:14	0	Layer saved exposure_summary
2017-05-23T15:38:14	0	The impact function could not run because of a bug.
2017-05-23T15:38:14	0	****Problem
			
			    The following problem(s) were encountered whilst running the analysis.
			
			 - TypeError
			
			****Suggestion
			
			    You can try the following to resolve the issue:
			
			 - Check that you have the latest version of InaSAFE installed - you may have encountered a bug that is fixed in a subsequent release.
			 - Check the InaSAFE documentation to see if you are trying to do something unsupported.
			 - Report the problem using the issue tracker at https://github.com/inasafe/inasafe/issues. Reporting an issue requires that you first create a free account at http://github.com. When you report the issue, please copy and paste the complete contents of this panel into the issue to ensure the best possible chance of getting your issue resolved.
			 - Try contacting one of the InaSAFE development team by sending an email to info@inasafe.org. Please ensure that you copy and paste the complete contents of this panel into the email.
			
			*****Details
			
			    These additional details were reported when the problem occurred.
			
			 - PyQt4.QtCore.QUrl(u'https://earthquake.usgs.gov/earthquakes/eventpage/us10008e3k#executive') is not JSON serializable
			
			*****Diagnostics
			
			 1. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/impact_function/impact_function.py", line 1083, in run self._run()
			 2. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/utilities/profiling.py", line 122, in with_profiling ret = fn(*args, **kwargs)
			 3. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/impact_function/impact_function.py", line 1279, in _run layer_purpose_exposure_summary['key'])
			 4. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/datastore/datastore.py", line 135, in add_layer KeywordIO().write_keywords(real_layer, layer.keywords)
			 5. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/utilities/keyword_io.py", line 170, in write_keywords write_iso19115_metadata(source, keywords)
			 6. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/utilities/metadata.py", line 68, in write_iso19115_metadata metadata.write_to_file(xml_file_path)
			 7. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/metadata/base_metadata.py", line 625, in write_to_file metadata = self.get_writable_metadata(file_format)
			 8. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/metadata/base_metadata.py", line 667, in get_writable_metadata metadata = self.xml
			 9. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/metadata/exposure_summary_layer_metadata.py", line 197, in xml root = super(ExposureSummaryLayerMetadata, self).xml
			 10. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/metadata/base_metadata.py", line 312, in xml elem.text = self.get_xml_value(name)
			 11. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/metadata/base_metadata.py", line 526, in get_xml_value return self.get_property(name).xml_value
			 12. In file "/Users/gavinfleming/.qgis2/python/plugins/inasafe/safe/metadata/property/dictionary_property.py", line 85, in xml_value string_value[k] = json.dumps(v)
			 13. In file "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 243, in dumps return _default_encoder.encode(obj)
			 14. In file "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 207, in encode chunks = self.iterencode(o, _one_shot=True)
			 15. In file "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 270, in iterencode return _iterencode(o, 0)
			 16. In file "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 184, in default raise TypeError(repr(o) + " is not JSON serializable")
			
```

**Comment by ismailsunni** [comment]
I tried to replicate this, but I can't.

I remember I have encountered this bug and I have fixed it.

**Comment by gubuntu** [comment]
closed via #4223
